Lib/test/support/os_helper.py

```diff
@@ -508,9 +535,38 @@ def temp_dir(path=None, quiet=False):
             rmtree(path)
 
 
+@contextlib.contextmanager
+def save_cwd(*, quiet=False):
+    """Return a context manager that restores the current working directory on
+    exit.
+
+    Prefer change_cwd() if you can. For most use cases it is cleaner.
```

Comment:
I confess I can't remember why I concluded `save_cwd()` was necessary for `test_permission_error_cwd()` (maybe a corner case on Windows?). So I've kicked off https://github.com/python/cpython/pull/124091 to shed light either way.

If #124091 shows `save_cwd()` isn't needed, then I'll remove it

Otherwise, I'll reply again with some clarifying questions. Thanks for the kick.

Reply:
Result: All tests passed on all platforms after deleting `save_cwd()` and using `change_cwd()` in https://github.com/python/cpython/actions/runs/10860873897.
Conclusion: `save_cwd()` is unnecessary for this PR. I'll remove it.